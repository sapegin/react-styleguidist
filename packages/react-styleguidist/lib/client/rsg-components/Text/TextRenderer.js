import "core-js/modules/es6.object.assign";
import "core-js/modules/web.dom.iterable";
import "core-js/modules/es6.array.iterator";
import "core-js/modules/es6.object.keys";
import "core-js/modules/es6.string.small";

import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import cx from 'classnames';

function _extends() { _extends = Object.assign || function (target) { for (let i = 1; i < arguments.length; i++) { const source = arguments[i]; for (const key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) {return {};} const target = _objectWithoutPropertiesLoose(source, excluded); let key; let i; if (Object.getOwnPropertySymbols) { const sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) {continue;} if (!Object.prototype.propertyIsEnumerable.call(source, key)) {continue;} target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) {return {};} const target = {}; const sourceKeys = Object.keys(source); let key; let i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) {continue;} target[key] = source[key]; } return target; }
export var styles = function styles(_ref) {
  const fontFamily = _ref.fontFamily;

      
const fontSize = _ref.fontSize;

      
const color = _ref.color;
  return {
    text: {
      fontFamily: fontFamily.base
    },
    inheritSize: {
      fontSize: 'inherit'
    },
    smallSize: {
      fontSize: fontSize.small
    },
    baseSize: {
      fontSize: fontSize.base
    },
    textSize: {
      fontSize: fontSize.text
    },
    baseColor: {
      color: color.base
    },
    lightColor: {
      color: color.light
    },
    em: {
      fontStyle: 'italic'
    },
    strong: {
      fontWeight: 'bold'
    },
    isUnderlined: {
      borderBottom: [[1, 'dotted', color.lightest]]
    }
  };
};
export function TextRenderer(_ref2) {
  let _cx;

  const classes = _ref2.classes;

      
const semantic = _ref2.semantic;

      
const size = _ref2.size;

      
const color = _ref2.color;

      
const underlined = _ref2.underlined;

      
const children = _ref2.children;

      
const props = _objectWithoutProperties(_ref2, ["classes", "semantic", "size", "color", "underlined", "children"]);

  const Tag = semantic || 'span';
  const classNames = cx(classes.text, classes["".concat(size, "Size")], classes["".concat(color, "Color")], (_cx = {}, _defineProperty(_cx, classes[semantic], semantic), _defineProperty(_cx, classes.isUnderlined, underlined), _cx));
  return React.createElement(Tag, _extends({}, props, {
    className: classNames
  }), children);
}
TextRenderer.propTypes = {
  classes: PropTypes.object.isRequired,
  semantic: PropTypes.oneOf(['em', 'strong']),
  size: PropTypes.oneOf(['inherit', 'small', 'base', 'text']),
  color: PropTypes.oneOf(['base', 'light']),
  underlined: PropTypes.bool,
  children: PropTypes.node.isRequired
};
TextRenderer.defaultProps = {
  size: 'inherit',
  color: 'base',
  underlined: false
};
export default Styled(styles)(TextRenderer);