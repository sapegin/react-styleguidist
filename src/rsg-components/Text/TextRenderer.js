import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import cx from 'classnames';

export const styles = ({ fontFamily, fontSize, color }) => ({
	text: {
		fontFamily: fontFamily.base,
		fontSize: fontSize.small,
		color: color.light,
	},
	isUnderlined: {
		borderBottom: [[1, 'dotted', color.lightest]],
	},
});

export function TextRenderer({ classes, children, underlined, ...other }) {
	const classNames = cx(classes.text, {
		[classes.isUnderlined]: underlined,
	});
	return <span className={classNames} {...other}>{children}</span>;
}

TextRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	children: PropTypes.node.isRequired,
	underlined: PropTypes.bool,
};

export default Styled(styles)(TextRenderer);
