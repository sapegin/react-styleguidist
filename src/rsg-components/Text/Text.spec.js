import React from 'react';
import { TextRenderer, styles } from './TextRenderer';

const props = {
	classes: classes(styles),
};

it('should render text', () => {
	const actual = shallow(<TextRenderer {...props}>Pizza</TextRenderer>);

	expect(actual).toMatchSnapshot();
});

it('should render underlined text', () => {
	const actual = shallow(<TextRenderer {...props} underlined>Pizza</TextRenderer>);

	expect(actual).toMatchSnapshot();
});

it('should render text with a title', () => {
	const actual = shallow(<TextRenderer {...props} title="Pasta">Pizza</TextRenderer>);

	expect(actual).toMatchSnapshot();
});
