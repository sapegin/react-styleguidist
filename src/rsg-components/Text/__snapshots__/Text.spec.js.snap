// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render text 1`] = `
<span
  className="text"
>
  Pizza
</span>
`;

exports[`should render text with a title 1`] = `
<span
  className="text"
  title="Pasta"
>
  Pizza
</span>
`;

exports[`should render underlined text 1`] = `
<span
  className="text isUnderlined"
>
  Pizza
</span>
`;
